src/components/card/safe-title.tsx

```diff
@@ -0,0 +1,19 @@
+const setSafeHeadingLevel = (h: string): any => {
+  const validHeadingLevels = ['h1', 'h2', 'h3', 'h4']
+  const safeHeading = h ? h.toLowerCase() : ''
+
+  return validHeadingLevels.includes(safeHeading) ? safeHeading : 'p'
+}
+const SafeTitle: React.FC<{headingLevel: string; className?: string}> = ({
+  children,
+  headingLevel,
+  className = '',
+}) => {
+  const Heading = setSafeHeadingLevel(headingLevel) as React.FC<{
+    children: React.ReactNode
+    className?: string
+  }>
+  return <Heading className={className}>{children}</Heading>
+}
+
+export default SafeTitle
```

Comment:
This is how I would build the component:

```jsx
import cx from 'classnames'

type Heading = React.ComponentPropsWithoutRef<any> & {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'p'
}

const SafeTitle = ({
  children,
  as,
  className,
}: Heading) => {
  const Component = as ?? 'p'
  return <Component className={cx(className)}>{children}</Component>
}
```

Reply:
`classnames` will apply the class if its passed in, if not it will return an empty string I believe

Reply:
Yeah that works too. I just like classnames because if you ever need to add default styles and combine the classes coming in from the props, it will do that for you. But it's definitely not needed here